src/routes/submit/submit.test.js

```diff
@@ -132,4 +152,109 @@ describe('submit route', () => {
     })
     expect(mockResponse.code).toHaveBeenCalledWith(statusCodes.ok)
   })
+
+  describe('file upload handling', () => {
+    it('should throw NotImplementedError if fileAnswer is skipped', async () => {
+      const skippedFileQuestion = {
+        question: 'upload-plan',
+        questionKey: 'upload-plan',
+        answer: {
+          type: 'file',
+          value: {
+            skipped: true
+          },
+          displayText: 'file.pdf'
+        }
+      }
+
+      const mockRequestSkippedFile = {
+        payload: {
+          sections: [
+            {
+              section: 'biosecurity-map',
+              sectionKey: 'biosecurity-map',
+              questionAnswers: [
+                emailQuestion,
+                fullNameQuestion,
+                skippedFileQuestion
+              ]
+            }
+          ]
+        }
+      }
+
+      await expect(
+        handler(mockRequestSkippedFile, mockResponse)
+      ).rejects.toThrow(NotImplementedError)
+    })
+
+    it('should return FILE_CANNOT_BE_DELIVERED if file size > 2MB and <= 10MB', async () => {
+      mockHandleUploadedFile.mockResolvedValue({
+        file: 'mock-file',
+        extension: '.pdf',
+        fileSizeInMB: 5
+      })
+
+      await handler(mockRequestWithFile, mockResponse)
+
+      expect(mockResponse.response).toHaveBeenCalledWith({
+        error: 'FILE_CANNOT_BE_DELIVERED'
+      })
+      expect(mockResponse.code).toHaveBeenCalledWith(
+        statusCodes.contentTooLarge
+      )
+    })
+
+    it('should return FILE_TOO_LARGE if file size > 10MB', async () => {
```

Comment:
"file size > 10MB at the point of upload"?

Reply:
done